### PR TITLE
Add DetachDatabaseCommand

### PR DESCRIPTION
**Pull request by ZacBlanco**
Removes a database from the AlluxioCatalog

After this PR we should revisit exactly what how we handle
accounting for mounted paths in the catalog service as well
as strategies to track and clean them up after detaching

### CODE REVIEW

File: table/server/master/src/main/java/alluxio/master/table/AlluxioCatalog.java [hunk 1]
```diff
@@ -246,6 +269,8 @@ public boolean processJournalEntry(Journal.JournalEntry entry) {
     } else if (entry.hasAddTable()) {
       Database db = mDBs.get(entry.getAddTable().getDbName());
       return db.processJournalEntry(entry);
+    } else if (entry.hasDetachDb()) {
+      apply(entry.getDetachDb());
```
Comment (yuzhu) [inline table/server/master/src/main/java/alluxio/master/table/AlluxioCatalog.java]: this method needs to return `true` if this journal entry has been handled.

File: table/server/master/src/main/java/alluxio/master/table/TableMaster.java [hunk 2]
```diff
@@ -37,6 +37,15 @@
   boolean attachDatabase(String dbName, String dbType, Map<String, String> configuration)
       throws IOException;
 
+  /**
+   * Remove an existing database in the catalog.
+   *
+   * @param dbName the database name to remove
+   * @return true if deletion is successful
+   */
+  boolean detachDatabase(String dbName)
```
Comment (calvinjia) [inline table/server/master/src/main/java/alluxio/master/table/TableMaster.java]: Is the database name a unique identifier?

**Comment by yuzhu** [comment]
@ZacBlanco also you need to add entries to `core/server/common/src/main/java/alluxio/master/journal/JournalEntryAssociation.java` and 
`core/server/common/src/test/java/alluxio/master/journal/JournalEntryAssociationTest.java`

File: table/server/master/src/main/java/alluxio/master/table/TableMaster.java (hunk 2 above)
Comment (ZacBlanco) [reply table/server/master/src/main/java/alluxio/master/table/TableMaster.java]: When attaching, the database name must be unique in the Alluxio master's catalog namespace - it's an entry in the `mDBs` map in `AlluxioCatalog.java`

**Comment by ZacBlanco** [comment]
jenkins, test this please

**Comment by ZacBlanco** [comment]
jenkins, test this please

File: table/server/master/src/main/java/alluxio/master/table/AlluxioCatalog.java [hunk 3]
```diff
@@ -238,6 +256,11 @@ private void apply(alluxio.proto.journal.Table.AttachDbEntry entry) {
     mDBs.put(dbName, db);
   }
 
+  private void apply(alluxio.proto.journal.Table.DetachDbEntry entry) {
+    String dbName = entry.getDbName();
+    mDBs.remove(dbName);
```
Comment (gpang) [inline table/server/master/src/main/java/alluxio/master/table/AlluxioCatalog.java]: how do ufs'es get unmounted from alluxio fs?

Comment (ZacBlanco) [reply table/server/master/src/main/java/alluxio/master/table/AlluxioCatalog.java]: we're going to leave them mounted

File: table/server/master/src/main/java/alluxio/master/table/AlluxioCatalog.java (hunk 1 above)
Comment (ZacBlanco) [reply table/server/master/src/main/java/alluxio/master/table/AlluxioCatalog.java]: done

**Comment by ZacBlanco** [comment]
alluxio-bot, merge this please

**Pull request merged**